File: 2335-minimum-amount-of-time-to-fill-cups/2335-minimum-amount-of-time-to-fill-cups.java
class Solution {
    public int fillCups(int[] amount) {
        int cold=amount[0];
        int warm=amount[1];
        int hot=amount[2];
        int total=cold+warm+hot;
        int maxVal=Math.max(cold,Math.max(warm,hot));
        return Math.max(maxVal,(total+1)/2);
    }
}

File: 2335-minimum-amount-of-time-to-fill-cups/2335-minimum-amount-of-time-to-fill-cups.cpp
class Solution {
public:
    int fillCups(vector<int>& amount) {
        int cold = amount[0];
        int warm = amount[1];
        int hot = amount[2];
        int total = cold + warm + hot;
        int maxVal = max({cold, warm, hot});
        return max(maxVal, (total + 1) / 2);
    }
};